### PR TITLE
implement salt.module.smartos_mdata

### PR DESCRIPTION
**Pull request by sjorge**
We should be able to manipulate mdata properties via salt.

``` bash
/opt/salt/bin/salt-call --local mdata.put salt 'is awesome'
[INFO    ] Executing command 'echo is awesome | /usr/sbin/mdata-put salt' in directory '/root'
```

``` yaml
local:
    True
```

``` bash
/opt/salt/bin/salt-call --local mdata.get smartos
[INFO    ] Executing command '/usr/sbin/mdata-list' in directory '/root'
[INFO    ] Executing command '/usr/sbin/mdata-get smartos' in directory '/root'
```

``` yaml
local:
    ----------
    smartos:
        is my OS of choice
```

``` bash
/opt/salt/bin/salt-call --local mdata.list
[INFO    ] Executing command '/usr/sbin/mdata-list' in directory '/root'
```

``` yaml
local:
    - salt
    - smartos
```

``` bash
/opt/salt/bin/salt-call --local mdata.delete salt
[INFO    ] Executing command '/usr/sbin/mdata-list' in directory '/root'
[INFO    ] Executing command '/usr/sbin/mdata-delete salt' in directory '/root'
```

``` yaml
local:
    ----------
    salt:
        True
```

### CODE REVIEW

**Comment by sjorge** [comment]
It would be nice to get everything, so added that too.

```bash
/opt/salt/bin/salt-call --local mdata.get
[INFO    ] Executing command '/usr/sbin/mdata-list' in directory '/root'
[INFO    ] Executing command '/usr/sbin/mdata-list' in directory '/root'
[INFO    ] Executing command '/usr/sbin/mdata-get smartos' in directory '/root'
[INFO    ] Executing command '/usr/sbin/mdata-list' in directory '/root'
[INFO    ] Executing command '/usr/sbin/mdata-get salt' in directory '/root'
```
```yaml
local:
    ----------
    salt:
        is awesome
    smartos:
        is my OS of choice
```

File: salt/modules/smartos_mdata.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    for k in keyname:
+        if mdata and k in valid_keynames:
+            cmd = '{0} {1}'.format(mdata, k)
+            ret[k] = __salt__['cmd.run_all'](cmd)['retcode'] == 0
+        else:
+            ret[k] = True
+
+    return ret
+
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4
```
Comment (rallytime) [inline salt/modules/smartos_mdata.py]: Can you remove this last line, please?

**Comment by rallytime** [comment]
@sjorge Awesome! This looks great. I only have one minor change to cleanup, which I left inline.

File: salt/modules/smartos_mdata.py (hunk 1 above)
Comment (sjorge) [inline salt/modules/smartos_mdata.py]: The '# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4' one?

Comment (rallytime) [inline salt/modules/smartos_mdata.py]: Ok. I just hadn't seen that before in salt. It doesn't hurt anything.

Comment (sjorge) [inline salt/modules/smartos_mdata.py]: Actually makes it a bit easier for vim users as it forced vim to use the tab settings specified in the file and not what the user has setup.

**Pull request merged**